### DISCUSSION TITLE
mnist example flatten

### DISCUSSION THREAD

**Discussion by deadsoul44** [google/flax#2533] [Q&A] [answered]
https://github.com/google/flax/blob/13fbace57ab3e1600aa3f8a2d596443c373f283d/examples/mnist/train.py#L47

The line above claims to flatten the array but it does nothing. 
```
x = jnp.ones((2, 3))
print(x)
# prints:
[[1. 1. 1.]
 [1. 1. 1.]]

y = x.reshape((x.shape[0], -1))
print(y)
# prints:
[[1. 1. 1.]
 [1. 1. 1.]]
```
Forgive me if I am missing something very obvious.

**Comment by cgarciae** [comment, marked as answer]
Hey @deadsoul44, it works if you have more than 2 dimension e.g:

```python
import jax.numpy as jnp

x = jnp.ones((8, 28, 28, 1))
y = x.reshape((x.shape[0], -1))
print(y.shape) # (8, 784)
```
else you the shape won't change as you note but all this is correct behaviour.

**Reply by cgarciae** [reply]
Note, flattening here means "flatten sample".

**Reply by deadsoul44** [reply]
Thanks a lot for the explanation.
